datasets/ar_sarcasm/README.md

```diff
@@ -0,0 +1,177 @@
+
+### Data Instances
+
+```javascript
+{'dialect': 1, 'original_sentiment': 0, 'sarcasm': 0, 'sentiment': 0, 'source': 'semeval', 'tweet': 'نصيحه ما عمرك اتنزل لعبة سوبر ماريو مش زي ما كنّا متوقعين الله يرحم ايامات السيقا والفاميلي #SuperMarioRun'}
+```
+
+### Data Fields
+
+- tweet: the original tweet text
+- sarcasm: 0 for non-sarcastic, 1 for sarcastic
+- sentiment: 0 for negative, 1 for neutral, 2 for positive
+- original_sentiment: 0 for negative, 1 for neutral, 2 for positive
+- source: the original source of tweet: SemEval or ASTD
+- dialect: 0 for Egypt, 1 for Gulf, 2 for Levant, 3 for Magreb, 4 for Modern Standard Arabic (MSA)
+
+### Data Splits
+
+The training set contains 8,437 tweets, while the test set contains 2,110 tweets.
+
+## Dataset Creation
+
+### Curation Rationale
+
+[More Information Needed]
+
+### Source Data
+
+#### Initial Data Collection and Normalization
+
+The dataset was created using previously available Arabic sentiment analysis datasets (SemEval 2017 and ASTD) and adds sarcasm and dialect labels to them.
+
+#### Who are the source language producers?
+
+SemEval 2017 and ASTD
+
+### Annotations
+
+#### Annotation process
+
+For the annotation process, we used Figure-Eight
+crowdsourcing platform. Our main objective was to annotate the
+data for sarcasm detection, but due to the challenges imposed by dialectal variations, we decided to add the annotation for dialects. We also include a new annotation for
+sentiment labels in order to have a glimpse of the variability and subjectivity between different annotators. Thus, the
+annotators were asked to provide three labels for each tweet
+as the following:
+
+- Sarcasm: sarcastic or non-sarcastic.
+- Sentiment: positive, negative or neutral.
+- Dialect: Egyptian, Gulf, Levantine, Maghrebi or Modern Standard Arabic (MSA).
+
+#### Who are the annotators?
+
+Figure-Eight crowdsourcing platform
+
+### Personal and Sensitive Information
+
+[More Information Needed]
+
+## Considerations for Using the Data
+
+### Social Impact of Dataset
+
+[More Information Needed]
+
+### Discussion of Biases
+
+[More Information Needed]
+
+### Other Known Limitations
+
+[More Information Needed]
+
+## Additional Information
+
+### Dataset Curators
+
+- Ibrahim Abu-Farha
+- Walid Magdy
+
+### Licensing Information
+
+MIT
+
+### Citation Information
+
+```
+@inproceedings{abu-farha-magdy-2020-arabic,
+    title = "From {A}rabic Sentiment Analysis to Sarcasm Detection: The {A}r{S}arcasm Dataset",
+    author = "Abu Farha, Ibrahim  and Magdy, Walid",
+    booktitle = "Proceedings of the 4th Workshop on Open-Source Arabic Corpora and Processing Tools, with a Shared Task on Offensive Language Detection",
+    month = may,
+    year = "2020",
+    address = "Marseille, France",
+    publisher = "European Language Resource Association",
+    url = "https://www.aclweb.org/anthology/2020.osact-1.5",
+    pages = "32--39",
+    language = "English",
+    ISBN = "979-10-95546-51-1",
+}
+```
```

Comment:
````suggestion
```

### Contributions

Thanks to [@mapmeld](https://github.com/mapmeld) for adding this dataset.
````

We added a new section :)